### ISSUE TITLE
YOLOE

### ISSUE THREAD

**Issue by HeavensDoor1** [ultralytics/ultralytics#19879]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Good afternoon, I have a question. I saw the latest news about YOLOE and would really like to test it on my dataset and classes. Is there a possibility to access the model publicly and see how it works, or is it not ready yet?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @HeavensDoor1, thank you for your interest in Ultralytics 🚀! We recommend exploring the [Docs](https://docs.ultralytics.com/) for new users, where you can find helpful resources, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions may already be answered there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a ❓ Question, as it seems, thank you for your query about YOLOE! Please provide as much additional information as possible, including specific details about your dataset or goals, to help us assist you better. Meanwhile, an Ultralytics engineer will review your question and provide further assistance soon 😊.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package, including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure you are using the latest features and updates:

```bash
pip install -U ultralytics
```

## Environments

YOLO models, including upcoming versions, can be run in any of the following up-to-date, verified environments (with all dependencies pre-installed):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
It's not available yet

**Comment by glenn-jocher** [comment]
@HeavensDoor1 yes, we are updating with YOLOE soon!

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged via [PR #19775](https://github.com/ultralytics/ultralytics/pull/19775), titled "ultralytics 8.3.99 New YOLOE Open-Vocabulary Models." This update introduces several exciting enhancements, including YOLOE models designed for open-vocabulary detection, segmentation, and advanced features like visual prompt embedding (SAVPE). Key updates relevant to this issue include:

- **YOLOE Integration**: New models designed for open-vocabulary detection and segmentation, supporting both text and visual prompts.
- **Improved Training and Validation Pipelines**: Specialized workflows for YOLOE, including prompt-free modes, enabling broader and more flexible use cases.
- **Documentation Enhancements**: Added detailed examples for training, validation, and prediction to ensure a smoother user experience.

Credit goes to the PR author @leonnil and contributors @jameslahm, @Laughing-q, @glenn-jocher, @wanjunling168, and @ilbash for their collaborative efforts in delivering these improvements. 

To test if these changes resolve the issue, consider updating to the latest version of the package:
```bash
pip install -U ultralytics>=8.3.99
```
Alternatively, install the latest commit from the main branch:
```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
You can also wait for the next official PyPI release if preferred.

Please let us know if this resolves your issue. Thank you for reporting and helping improve the project! 🙏 If the issue persists, feel free to share further details, and we’ll gladly assist.

**Comment by glenn-jocher** [comment]
Great news @HeavensDoor1! YOLOE is now officially available in ultralytics 8.3.99. You can test it on your dataset using both text and visual prompts. YOLOE supports instance segmentation out of the box and is designed for open-vocabulary detection - meaning it can detect objects not seen during training.

For a quick start with your own dataset:
```python
from ultralytics import YOLO

# Load the model
model = YOLE("yoloe-11s-seg.pt")  # Choose from various sizes (11s, 11m, 11l)

# Train on your dataset
model.train(data="path/to/your/data.yaml", epochs=50, imgsz=640)

# For inference with specific classes
model.set_classes(["your_class1", "your_class2"])
results = model.predict("your_image.jpg")
```

The model comes in different sizes and supports both prompt-free and prompted detection. You can find comprehensive documentation at our [YOLOE page](https://docs.ultralytics.com/models/yoloe/).
